modules/features/search/src/main/java/au/com/shiftyjelly/pocketcasts/search/SearchInlineResultsPage.kt

```diff
@@ -130,13 +130,13 @@ private fun SearchResultsView(
             }
         }
     }
-    val podcasts by remember { mutableStateOf(state.podcasts) }
     val podcastsRowState = rememberLazyListState()
     val episodesRowState = rememberLazyListState()
-    LaunchedEffect(key1 = state.podcasts) {
-        // Only reset the scroll state if we have podcasts and they are different. This prevents resetting the
-        // scroll state when navigating back to the search results after tapping on one of the results.
-        if (state.podcasts.isNotEmpty() && state.podcasts != podcasts) {
+    val initialPodcasts by remember(state.podcasts.isEmpty()) { mutableStateOf(state.podcasts) }
```

Comment:
The expression `state.podcasts.isEmpty()` is not a reliable key. For instance, if I type "redwood," the app will store a list of podcasts. However, when I delete characters to leave only "red," the original list of podcasts is still retained.

A better approach would be to use `state.podcasts.map { it.uuid }` as a `val`. It is already created during every recomposition for the `LaunchedEffect` anyway. Then it can be used inside the `LaunchedEffect` for comparison to control scrolling.